sql/core/src/main/scala/org/apache/spark/sql/execution/window/WindowFunctionFrame.scala

```diff
@@ -201,7 +201,11 @@ class FrameLessOffsetWindowFunctionFrame(
   override def prepare(rows: ExternalAppendOnlyUnsafeRowArray): Unit = {
     resetStates(rows)
     if (ignoreNulls) {
-      findNextRowWithNonNullInput()
+      if (Math.abs(offset) > rows.length) {
```

Comment:
We no need `Math.abs(offset)` here.
`findNextRowWithNonNullInput` skips all the rows if `offset < 0`

Reply:
I think the current code is clear. It's not obvious that `findNextRowWithNonNullInput` works for negative offsets, and I don't think perf is a problem here.